packages/ui/src/types/authenticator/form.ts

```diff
@@ -64,6 +64,16 @@ export interface FormFieldOptions {
   type?: string;
   /** Desired autocomplete HTML attribute */
   autocomplete?: string;
+  /** Desired HTML pattern type */
+  pattern?: string | undefined;
```

Comment:
Not sure if this going to work as intended without typing as `RegExp` :
<img width="821" alt="CleanShot 2023-08-24 at 09 06 07@2x" src="https://github.com/aws-amplify/amplify-ui/assets/15002885/9306c56e-d16e-4496-822f-e0389d8f9b5c">

Reply:
Hm I tried testing with a value like this and it seemed to work, open to other ideas here though!

```
given_name: {
      isRequired: true,
      label: 'custom name:',
      pattern: "[Bb]anana|[Cc]herry",
    },
```

Reply:
But yes we will have an issue where you cannot pass a plain regexp here

Reply:
Looked at the base input typing and are with just `string`